services/api/src/routes/categories.js

```diff
@@ -14,39 +15,25 @@ router
     validate({
       body: Joi.object({
         name: Joi.string(),
-        ids: Joi.array().items(Joi.string()),
-        skip: Joi.number().default(0),
-        sort: Joi.object({
-          field: Joi.string().required(),
-          order: Joi.string().valid('asc', 'desc').required(),
-        }).default({
-          field: 'createdAt',
-          order: 'desc',
-        }),
-        limit: Joi.number().positive().default(50),
+        ...searchValidation(),
       }),
     }),
     async (ctx) => {
-      const { ids = [], sort, skip, limit, name } = ctx.request.body;
-      const query = {
-        ...(ids.length ? { _id: { $in: ids } } : {}),
-        ...(name ? { name: { $regex: name, $options: 'i' } } : {}),
-        deletedAt: { $exists: false },
-      };
-
-      const data = await Category.find(query)
-        .sort({ [sort.field]: sort.order === 'desc' ? -1 : 1 })
-        .skip(skip)
-        .limit(limit);
-
-      const total = await Category.countDocuments(query);
+      const { body } = ctx.request;
+      const { name } = body;
+      const query = getSearchQuery(body, {
+        keywordFields: ['name'],
```

Comment:
I think we need to be build out search functions with $regex for the name

I am thinking something like 

```
getSearchQuery(body, {
        keywordFields: [{ field: 'name', type: 'search' }, "otherField"],
})
```

{ field: 'name', type: 'search' } => would translate to the $regexp mongodb query

Reply:
actually I went ahead and `keywordFields` a regexp (have a look at `utils/search`). I prefer this as a `keyword` to me implies greediness... it should match whatever it can

Reply:
just updated per feedback